### ISSUE TITLE
Problems with ListResult.php

### ISSUE THREAD

**Issue by uuf6429** [chargebee/chargebee-php#41]
Let's take a look at [`ListResult.php`](https://github.com/chargebee/chargebee-php/blob/master/lib/ChargeBee/ListResult.php).

Here are the key facts:
- it defines `_items` and `_index` properties (by the way, underscore is unnecessary)
- after construction / initialisation, it fills up `_items` with the result records
- checking (`offsetExists`), setting (`offsetSet`) and removing (`offsetUnset`) offsets affects arbitrary properties
- getting an offset (`offsetGet`) access the undefined/magic `list` property
- looping (`current` / `valid`) actually works with `_items` property

So from my understand:
```php
foreach($result as $k => $v){
    echo "$k=$v"; // produces for example 0=somevalue
}

echo "0={$result[0]}"; // produces 0=null instead of 0=somevalue
```

As I see from `lib/ChargeBee/Request.php:39`, response originally contains items inside `list` index....so to sum this all up:
- the `response` property does not contain the original response, but rather the value of `response['list']`
- `offset*` methods are buggy (they either get arbitrary properties or for `offsetGet`, the non-existent 'list' property)

**Comment by uuf6429** [comment]
By the way, a lot of code from that class can be removed if you make it extend `\ArrayObject'. Here's an example: https://3v4l.org/PvFLQ (works at least since PHP 5.4)

**Comment by uuf6429** [comment]
Just realised, the `offsetGet` part of the problem mentioned above has been fixed in #22 (not released since 2018).

**Comment by cb-khushbubibay** [comment]
This issue has been addressed in v3.4.0. Hence closing it.
